test/system/inspect.spec.ts

```diff
@@ -2,35 +2,102 @@ import { inspect } from '../../lib';
-
-    const result = await inspect('.', FILENAMES.poetry.lockfile);
-    expect(result).toMatchObject({
-      plugin: {
-        name: 'snyk-python-plugin',
-        runtime: expect.any(String), // any version of Python
-        targetFile: FILENAMES.poetry.lockfile,
-      },
-      package: null, // no dep-tree
-      dependencyGraph: {}, // match any dep-graph (equality checked below)
+  const originalCurrentWorkingDirectory = process.cwd();
+
+  afterEach(() => {
+    process.chdir(originalCurrentWorkingDirectory);
+  });
+
+  describe('poetry projects', () => {
+    it('should return expected dependencies for poetry-app', async () => {
+      const workspace = 'poetry-app';
+      chdirWorkspaces(workspace);
+
+      const result = await inspect('.', FILENAMES.poetry.lockfile);
+      expect(result).toMatchObject({
+        plugin: {
+          name: 'snyk-python-plugin',
+          runtime: expect.any(String), // any version of Python
+          targetFile: FILENAMES.poetry.manifest,
+        },
+        package: null, // no dep-tree
+        dependencyGraph: {}, // match any dep-graph (equality checked below)
+      });
+
+      const builder = new DepGraphBuilder(
+        { name: 'poetry' },
+        { name: 'poetry-fixtures-project', version: '0.1.0' }
+      );
+      const expected = builder
+        .addPkgNode({ name: 'jinja2', version: '2.11.2' }, 'jinja2')
+        .connectDep(builder.rootNodeId, 'jinja2')
+        .addPkgNode({ name: 'markupsafe', version: '1.1.1' }, 'markupsafe')
+        .connectDep('jinja2', 'markupsafe')
+        .build();
+
+      expect(result.dependencyGraph).toEqualDepGraph(expected);
     });
+  });
+
+  it('should return correct target file for poetry project when relative path to poetry lock file is passed', async () => {
+    const dirname = 'test/fixtures/poetry-project';
+    const manifestFilePath = `${dirname}/poetry.lock`;
+
+    const result = await inspect('.', manifestFilePath);
+
+    const expectedTargetFile = `${dirname}/pyproject.toml`;
+    expect(result.plugin.targetFile).toEqual(expectedTargetFile);
+  });
 
-    const builder = new DepGraphBuilder(
-      { name: 'poetry' },
-      { name: 'poetry-fixtures-project', version: '0.1.0' }
-    );
-    const expected = builder
-      .addPkgNode({ name: 'jinja2', version: '2.11.2' }, 'jinja2')
-      .connectDep(builder.rootNodeId, 'jinja2')
-      .addPkgNode({ name: 'markupsafe', version: '1.1.1' }, 'markupsafe')
-      .connectDep('jinja2', 'markupsafe')
-      .build();
-
-    expect(result.dependencyGraph).toEqualDepGraph(expected);
+  describe('Pipfile projects', () => {
+    const mockedExecuteSync = jest.spyOn(subProcess, 'executeSync');
+    const mockedExecute = jest.spyOn(subProcess, 'execute');
+
+    afterEach(() => {
+      mockedExecuteSync.mockClear();
+      mockedExecute.mockClear();
+    });
+
+    it('should return correct target file for pipenv project when relative path to pipfile lock file is passed', async () => {
```

Comment:
with additional of this test, do you think that we can remove this [tap](https://github.com/snyk/snyk-python-plugin/blob/d1866c46c0d53988a7b6bbbb4a92ce7e2a2d36b4/test/system/inspect.test.js#L925-L929) test?